hlx_statics/scripts.js

```diff
@@ -1,17 +1,64 @@
 let $IS_HLX_PATH = false;
 let $IS_STAGE = false;
+let $IS_DEV = false;
 
-if(window.location.host.indexOf('hlx.page') >= 0 || window.location.host.indexOf('hlx.live') >= 0 || window.location.host.indexOf('localhost') >= 0){
+if (window.location.host.indexOf('hlx.page') >= 0 || window.location.host.indexOf('hlx.live') >= 0 || window.location.host.indexOf('localhost') >= 0) {
   $IS_HLX_PATH = true;
 }
 
-if(window.location.host.indexOf('stage.adobe.io') >= 0 || window.location.host.indexOf('developer-stage') >= 0){
+if (window.location.host.indexOf('stage.adobe.io') >= 0 || window.location.host.indexOf('developer-stage') >= 0) {
   $IS_STAGE = true;
 }
 
+if (window.location.host.indexOf('localhost') >= 0) {
+  $IS_DEV = true;
+}
+
 window.adobeid = {};
 
-if($IS_HLX_PATH) {
+const setQueryStringParameter = (name, value) => {
+  const params = new URLSearchParams(window.location.search);
+  params.set(name, value);
+  window.history.replaceState({}, "", `${window.location.pathname}?${params}`);
+};
+const getQueryString = () => {
+  const params = new URLSearchParams(window.location.search);
+  return params.toString();
+}
+const setExpectedOrigin = () => {
+  if ($IS_DEV) {
+    return 'http://localhost:8000';
+  } else if ($IS_STAGE || $IS_HLX_PATH) {
+    return 'https://developer-stage.adobe.com';
+  } else {
+    return 'https://developer.adobe.com';
+  }
+}
+const setSearchFrameSource = () => {
+  const src = $IS_DEV ? setExpectedOrigin() : `${setExpectedOrigin()}/search-frame`;
+  const queryString = getQueryString();
+  if (queryString) {
+    return `${src}?${queryString}`;
+  } else {
+    return src;
+  }
+}
```

Comment:
As mentioned in the pre-commit comment, 

the `message` event's expected origin has been refactored into a helper function `setExpectedOrigin` with IFs over ternary conditionals for readability. The search iframe source has also been refactored and now utilizes `setExpectedOrigin` for base url.

I've also realized there was existing stage/helix context flags so I've added a standalone `$IS_DEV` flag and utilized them in the new functions.

Reply:
@timkim can you go over this comment

Reply:
I think it's fine to have `$IS_DEV` as is. There are some cases where we need to know if it's localhost or not and `$IS_DEV` does that.

Reply:
Yep but I was wondering if checking for localhost is needed in the $IS_HLX_PATH.

Currently both $IS_DEV and $IS_HLX_PATH will be true on localhost.